pipelines/nyu/adapters/course_catalog/src/process_data.rb

```diff
@@ -1,60 +1,101 @@
 require 'pry'
+require 'uri'
 
 # List of courses for a subject
 def extract_listings html
```

Comment:
It is best practice not to create things in the global namespace (without a good reason). All code should be contained in classes or modules. If you want to keep this code in its own file consider wrapping it in a module.

Reply:
Ok, I put the code in a module called `ExtractListingData`